src/Transport/Topology/ICreateTopology.cs

```diff
@@ -14,8 +14,10 @@ public interface ICreateTopology
     }
 
     // TODO: Move into internalized ICreateTopology in v8
-    interface ITearDownTopology
+    interface ICreateTopologyInternal : ICreateTopology
```

Comment:
My problem was the following:

We had `ITeardownTopology` which was introduced for internal usage until we can internalise the interfaces. But having an AssertManagedRights doesn't really make sense on `ITeardownTopology`. That's why I went with introducing this internal interface (should also exist in develop) that now contains the methods defined in the public base interface as well as the assert method.